common/src/api/external/error.rs

```diff
@@ -255,32 +255,66 @@ impl From<Error> for HttpError {
     }
 }
 
-impl<T: std::fmt::Debug> From<progenitor::progenitor_client::Error<T>>
+pub trait ClientError: std::fmt::Debug {
+    fn message(&self) -> String;
+}
+
+// TODO this From may give us a shortcut in situtations where we want more
+// robust consideration of errors. For example, while some errors from other
+// services may directly result in errors that percolate to the external
+// client, others may required, for example, retries with an alternate service
+// instance or additional interpretation to sanitize the output error.
+//
+// ***********************************************************************
+// * THIS FROM SHOULD BE REMOVED PRIOR TO SHIPPING TO AVOID LEAKING DATA *
```

Comment:
I've been using `TODO-security` for stuff like this.

Reply:
much terser